rapids-cmake/cpm/detail/generate_patch_command.cmake

```diff
@@ -0,0 +1,102 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cpm_generate_patch_command
+---------------------------------
+
+.. versionadded:: v22.10.00
+
+Applies any relevant patches to the provided CPM package
+
+.. code-block:: cmake
+
+  rapids_cpm_generate_patch_command(<pkg> <version> patch_command)
+
+#]=======================================================================]
+# cmake-lint: disable=R0915,E1120
+function(rapids_cpm_generate_patch_command package_name version patch_command)
+  list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cpm.generate_patch_command")
+
+  include("${rapids-cmake-dir}/cpm/detail/load_preset_versions.cmake")
+  rapids_cpm_load_preset_versions()
+
+  include("${rapids-cmake-dir}/cpm/detail/get_default_json.cmake")
+  include("${rapids-cmake-dir}/cpm/detail/get_override_json.cmake")
+  get_default_json(${package_name} json_data)
+  get_override_json(${package_name} override_json_data)
+
+  string(JSON json_data ERROR_VARIABLE no_default_patch GET "${json_data}" patches)
+  string(JSON override_json_data ERROR_VARIABLE no_override_patch GET "${override_json_data}"
+         patches)
+  if(no_default_patch AND no_override_patch)
+    return() # no patches
+  endif()
+  if(NOT no_override_patch)
+    set(json_data "${override_json_data}")
+  endif()
+
+  # Parse required fields
+  function(rapids_cpm_json_get_value jdata name)
```

Comment:
Does `jdata` stand for `json_data`? Would a more explicit name like `json_data` shadow the `json_data` in the enclosing scope? Not sure how scopes work in CMake. Maybe `json_data_` or something like that.

Reply:
Yeah it would cause shadowing, and cmake-lint will get angry :)

I can do `json_data_`